Link failure on 64-bit Gentoo

I got the following error when building helloworld.sml, and no final executable is produced

```
$HOME/MLKIT/bin/mlkit helloworld.sml

[reading source file:	helloworld.sml]
[wrote X64 code file:	MLB/RI_GC/helloworld.sml.s]
[wrote X64 code file:	MLB/RI_GC/base-link_objects.s]
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: MLB/RI_GC/base-link_objects.o: relocation R_X86_64_32S against `.data' can not be used when making a PIE object; recompile with -fPIC
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /home/chris/MLKIT/lib/mlkit/basis/MLB/RI_GC/Initial.sml.o: relocation R_X86_64_32S against `.text' can not be used when making a PIE object; recompile with -fPIC
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /home/chris/MLKIT/lib/mlkit/basis/MLB/RI_GC/General.sml.o: relocation R_X86_64_32S against `.text' can not be used when making a PIE object; recompile with -fPIC
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /home/chris/MLKIT/lib/mlkit/basis/MLB/RI_GC/Initial2.sml.o: relocation R_X86_64_32S against `.text' can not be used when making a PIE object; recompile with -fPIC
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: ./MLB/RI_GC/helloworld.sml.o: relocation R_X86_64_32S against `.text' can not be used when making a PIE object; recompile with -fPIC
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: final link failed: nonrepresentable section on output
collect2: error: ld returned 1 exit status
[wrote executable file:	run]
```

This is on 64-bit Gentoo, using a plain 17.0 profile. I ran the following to install

```
git clone https://github.com/melsman/mlkit.git
./autobuild
./configure --prefix=$HOME/MLKIT
make mlkit
make mlkit_libs
make install
```

Let me know what other information you would need

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,5 +1,7 @@
 language: c
 
+dist: xenial
+
 os:
  - linux
  - osx

--- a/src/Compiler/Backend/X64/ExecutionX64.sml
+++ b/src/Compiler/Backend/X64/ExecutionX64.sml
@@ -68,15 +68,16 @@ structure ExecutionX64: EXECUTION =
     val _ = Flags.add_string_entry
 	let val macgcc = "gcc -Wl,-no_pie,-stack_size,0x10000000,-stack_addr,0xc0000000"
 	    val gcc = if onmac_p() then macgcc
-		      else "gcc"
+		      else "gcc -no-pie"
 	in
 	    {long="c_compiler", short=SOME "cc", item=ref gcc,
 	     menu=["Control", "C compiler (used for linking)"],
 	     desc="This option specifies which C compiler is\n\
 		  \used for linking. When linking with c++\n\
 		  \libraries, 'g++' is the linker you want.\n\
-		  \On Linux the default is 'gcc', whereas on\n\
-                  \Mac OS X, the default is '" ^ macgcc ^ "'."}
+		  \On Linux the default is 'gcc -no-pie',\n\
+                  \whereas on macOS, the default is\n\
+                  \'" ^ macgcc ^ "'."}
 	end
 
     val _ = Flags.add_string_entry
@@ -87,7 +88,7 @@ structure ExecutionX64: EXECUTION =
 	    {long="assembler", short=SOME "as", item=ref ass,
 	     menu=["Control", "Assembler command"],
 	     desc="This option specifies the assembler used.\n\
-		  \On Linux the default is '" ^ linux_as ^ "'. On Mac OS X,\n\
+		  \On Linux the default is '" ^ linux_as ^ "'. On macOS,\n\
                   \the default is '" ^ mac_as ^ "'."}
 	end
 

